_data-prepper/pipelines/configuration/sources/kafka.md

```diff
@@ -0,0 +1,145 @@
+`topics` | Yes | JSON array | The topics inside of Kafka where the Data Prepper `kafka` source reads messages. You can configure up to 10 topics. For more information about options you need to configure inside `topics`, see [Topics](#topics).
+`schema` | No | JSON array | The schema registry configuration. For more information, see [Schema](#schema)
+`authentication` | No | JSON array | Set the authentication options for both the pipeline and Kafka. For more information, see [Authentication](#authentication).
+`encryption` | No | JSON array | The encryption configuration. For more information, see [Encryption](#encryption).
+`aws` | No | JSON array | The AWS configuration. See [aws](#aws) for details.
+`acknowledgments` | No | Boolean | If `true`, enables the `kafka` source to receive [end-to-end acknowledgments]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/pipelines/#end-to-end-acknowledgments) when events are received by OpenSearch sinks. Default is `false`.
+`acknowledgements_timeout` | No | Time | The maximum time to wait for acknowledgements to be received. Default is `30s`.
+`client_dns_lookup` | Yes, when a DNS alias is used. | String | Sets Kafka's `client.dns.lookup` option. Default is `default`.
+
+### Topics
+
+Use the following options in the `topics` array.
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`name` | Yes | String | The name of each Kafka topic.
+`group_id` | Yes | String | Sets Kafka's `group.id` option.
+`workers` | No | Integer | The number of multithreaded consumers associated with each topic. Default is `2`. The maximum value is `200`.
+`serde_format` | No | String | Indicates the serialization and the deserialization format of the messages in the topic. Default is `plaintext`.
+`auto_commit` | No | Boolean | When `false`, the consumer's offset will no be periodically committed to Kafka in the background. Default is `false`.
+`commit_interval` | No | Integer | When `auto_commit` is set to `true`, sets the frequency in seconds that the consumer offsets are auto-committed to Kafka through Kafka's `auto.commit.interval.ms` option. Default is `5s`.
+`session_timeout` | No | Integer | The amount of time for the source to detect client failures when using Kafka's group management features, which can be used to balance the data stream. Default is `45s`.
+`auto_offset_reset` | No | String | Automatically resets the offset to the earlier or latest offset through Kafka's `auto.offset.reset` option. Default is `latest`.
+`thread_waiting_time` | No | Integer | The time that threads wait until the preceding thread completes its task and signals the next thread. The Kafka consumer API poll timeout value is set to half of this setting. Default is `5s`.
+`max_partition_fetch_bytes` | No | Integer | Sets the maximum limit in bytes for max data returns from each partition through Kafka's `max.partition.fetch.bytes`. Default is `1048676`.
```

Comment:
A word should follow the second instance of `max.partition.fetch.bytes`. "option"? "setting"?

Reply:
setting.